uctMaths/competition/compadmin.py

```diff
@@ -630,6 +634,51 @@ def export_awards(request):
     output_workbook.save(response)
     return response
 
+def makeCertificates(students, assigned_school):
+    
+    certPath = "../../Certificates/"
+
+    if len(students) > 0:
+        awardCerts = {"G": "goldTemplate.docx", "M": "meritTemplate.docx", None: "participationTemplate.docx", "MOX": "meritOxfordTemplate.docx", "OX": "oxfordTemplate"}
+
+        try:
+            os.mkdir(str(students[0].school).replace(" ", "_") + "_Certificates")
+    
+        finally:
+            certs = []
+            filename = str(str(students[0].school).replace(" ", "_") + "_Certificates")
```

Comment:
Also why is this in a finally block?

Reply:
os.mkdir throws an exception when the directory already exists, so I had to put it in a try block, which then complained unless I put the rest of the code in a finally block. If mkdir is replaced by the temp directory you recommended in the other comment, it should probably be fine to remove this

Reply:
I'd prefer for you to use a temporary directory here, but for future reference: When code throws an exception that you want to ignore you usually write it like this:

``` python
try:
    #  <<Code that throws an exception that you want to ignore>>
except Exception:
    pass
# <<Rest of the code>>
```

Not like this:

``` python
try:
    # <<Code that throws an exception you want to ignore>>
finally:
    # <<Rest of the code>>
```

`finally` blocks are usually used for cleanup. The former way to write the code is clearer because you are catching the exception and then letting it pass.

The latter way to write the code is weird, because the exception still propagates even after the finally code is executed.

Reply:
Thanks for the tip! I've removed it now, as it's not needed for the temporary file